internal/controller/workloads/rolebasedgroup_controller.go

```diff
@@ -341,6 +370,41 @@ func (r *RoleBasedGroupReconciler) ReconcileScalingAdapter(
 
 	return r.client.Create(ctx, rbgScalingAdapter)
 }
+func (r *RoleBasedGroupReconciler) getOrCreateRevisionIfNotExist(ctx context.Context, rbg *workloadsv1alpha1.RoleBasedGroup) (*appsv1.ControllerRevision, error) {
+	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			workloadsv1alpha1.SetNameLabelKey: rbg.Name,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	revisions, err := utils.ListRevisions(ctx, r.client, rbg, selector)
+	if err != nil {
+		return nil, err
+	}
+	revisions, err = utils.CleanExpiredRevision(ctx, r.client, rbg, revisions)
```

Comment:
I suggest moving the cleanup logic for expired revisions (CleanExpiredRevision) into its own step within the reconcile loop; do not embed it inside getOrCreateRevisionIfNotExist.

Reply:
Just like https://github.com/kubernetes-sigs/lws/blob/main/pkg/controllers/leaderworkerset_controller.go#L179